benchmark/Benchmark/Compare.hs

```diff
@@ -0,0 +1,45 @@
+module Benchmark.Compare
+       (main
+       ) where
+
+import Data.Text (Text)
+import Gauge.Main (bench, bgroup, defaultMain, whnf)
```

Comment:
You should use `nf` everywhere here instead of `whnf`. Because data types are very nested, it's not enough to evaluate only to weak-head-normal form, you need to evaluate to normal form.